types/config.go

```diff
@@ -36,6 +44,80 @@ type ConfigFile struct {
 	RootFS     v1.RootFS  `json:"rootfs"`
 }
 
+// IOTypes represents the input and output types a model can handle
+type IOTypes struct {
+	Input  []string `json:"input" validate:"dive,oneof=text embedding image audio video"`
+	Output []string `json:"output" validate:"dive,oneof=text embedding image audio video"`
+}
+
+// Validate validates the IOTypes
+func (io *IOTypes) Validate() error {
+	validTypes := map[string]bool{
+		IOTypeText:      true,
+		IOTypeEmbedding: true,
+		IOTypeImage:     true,
+		IOTypeAudio:     true,
+		IOTypeVideo:     true,
+	}
+
+	// Check for duplicates and validate input types
+	seen := make(map[string]bool)
+	for _, t := range io.Input {
+		t = strings.TrimSpace(t)
+		if !validTypes[t] {
+			return fmt.Errorf("invalid input type: %s", t)
+		}
+		if seen[t] {
+			return fmt.Errorf("duplicate input type: %s", t)
+		}
+		seen[t] = true
+	}
+
+	// Check for duplicates and validate output types
+	seen = make(map[string]bool)
+	for _, t := range io.Output {
+		t = strings.TrimSpace(t)
+		if !validTypes[t] {
+			return fmt.Errorf("invalid output type: %s", t)
+		}
+		if seen[t] {
+			return fmt.Errorf("duplicate output type: %s", t)
+		}
+		seen[t] = true
+	}
+
+	return nil
+}
+
+// Capabilities describes what the model can do
+type Capabilities struct {
+	IO        IOTypes `json:"io"`
```

Comment:
nit: instead of `io` maybe `modalities`? I like the grouping, but `io` isn't really adding additional context when the two fields are `input` and `output`, whereas `modalities` communicates what exactly about the input and output is being described.

wdyt @bergwolf? Since we are aiming for convergence, consensus is probably more important than the exact keys here.  I am also okay with `input_types` and `output_types` (without the additional layer of nesting) like in https://github.com/CloudNativeAI/model-spec/pull/52 if there is a strong preference